wisp-lease/src/main/kotlin/wisp/lease/Extensions.kt

```diff
@@ -0,0 +1,24 @@
+package wisp.lease
+
+/** Converts a [lease] into an [AutoCloseable] resource. */
+class AutoCloseableLease(private val lease: Lease): Lease by lease, AutoCloseable {
+  override fun close() {
+    lease.release()
+  }
+}
+
+/**
+ * Attempts to acquire an [AutoCloseableLease].
+ *
+ * Use like
+ *
+ * ```
+ * leaseManager.acquireOrNull("some-lease")?.use { lease ->
+ *   // Do something with the lease.
+ * }
+ * ```
+ */
+fun LeaseManager.acquireOrNull(name: String): AutoCloseableLease? {
```

Comment:
One potential downside of this is it's still possible to acquire and then forget to close a lease. I've seen the `useX` convention used in the kotlin stdlib that avoids this by just accepting a lambda directly. Example that jumps to mind is the [useLines](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/java.io.-file/use-lines.html) method. 
Eg: 
```kotlin
File("").useLines { lines -> println(lines) }
```

This may be a bit awkward her - I'm not sure what the behavior should be if you can't acquire ...

Reply:
You raise a good point, which is also still a problem with the prescribed usage without this extension. 
Leases are held only for some time, so if a user forgets to close the lease, it will be let go eventually.

Something like `leaseManager.useLease { lease -> ... }` would be nicer from that perspective, however it would probably have to come with the stipulation that the lambda doesn't run if the lease can't be acquired. Unsure if that would be surprising. Perhaps `leaseManager.maybeUseLease { lease -> ... }`?  Dunno! 🤷